script/chart-museum.sh

```diff
@@ -88,17 +79,47 @@ installChartMuseum() {
     --set env.secret.BASIC_AUTH_PASS=$CHARTMUSEUM_PWD
   info "Waiting for ChartMuseum to be ready..."
   kubectl rollout status -w deployment/chartmuseum --namespace=${CHARTMUSEUM_NS}
+  
+  echo "Installing Ingress for ChartMuseum with access through host ${CHARTMUSEUM_HOSTNAME}"
+  kubectl create -n $CHARTMUSEUM_NS -f - -o yaml << EOF
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    nginx.ingress.kubernetes.io/connection-proxy-header: keep-alive
+    nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
+    nginx.ingress.kubernetes.io/proxy-buffer-size: "8k"
+    nginx.ingress.kubernetes.io/proxy-buffers: "4.0"
+  name: chartmuseum
+spec:
+  ingressClassName: nginx
+  rules:
+  - host: ${CHARTMUSEUM_HOSTNAME}
+    http:
+      paths:
+      - backend:
+          service:
+            name: chartmuseum
+            port:
+              number: 8080
+        path: /
+        pathType: ImplementationSpecific
+EOF
+  sleep 20
```

Comment:
Why this manual sleep here? Maybe it's for giving some time to the ingress controller to become available? If so, why not just sth like: `        kubectl wait --namespace ingress-nginx --for=condition=ready pod --selector=app.kubernetes.io/component=controller --timeout=120s
`

Reply:
That `sleep` is the key that missing. Before that, `Ingress` wasn't ready by the time it was used some instructions after. I thought about adding the `kubectl wait`, but that is for the nginx controller itself which is ready way before ([when the cluster is installed](https://github.com/vmware-tanzu/kubeapps/blob/bfec7d10a1d87e2b7808e3244bada03e7fa40a99/.circleci/config.yml#L364)).
It seems that it takes some time for the route to be processed by the Nginx controller?